server/src/main/java/org/apache/iotdb/db/engine/storagegroup/timeindex/ITimeIndex.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.engine.storagegroup.timeindex;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.Set;
+import org.apache.iotdb.db.exception.PartitionViolationException;
+
+public interface ITimeIndex {
+
+  /**
+   * init startTimes with Long.MAX_VALUE, endTimes with Long.MIN_VALUE
+   */
+  void init();
+
+  /**
+   * serialize to outputStream
+   *
+   * @param outputStream outputStream
+   */
+  void serialize(OutputStream outputStream) throws IOException;
+
+  /**
+   * deserialize from inputStream
+   *
+   * @param inputStream inputStream
+   * @return TimeIndex
+   */
+  ITimeIndex deserialize(InputStream inputStream) throws IOException;
+
+  /**
+   * deserialize from byte buffer
+   *
+   * @param buffer byte buffer
+   * @return TimeIndex
+   */
+  ITimeIndex deserialize(ByteBuffer buffer);
+
+  /**
+   * called by TsFileResource.close()
+   */
+  void close();
```

Comment:
This method is a bit confusing. What is this method used for? Or what should the caller expect when calling it?

Reply:
TimeIndex may do something (for example, trim start time array and end time array) when TsFileResource is closing. I updated the java doc.